src/main/java/io/cdap/delta/datastream/util/Utils.java

```diff
@@ -995,4 +1008,25 @@ public Credentials getCredentials() throws IOException {
       }
     }).build());
   }
+
+  private static Optional<ApiException> getApiExceptionFromCauses(Throwable e) {
+    int maxLevels = 5;
```

Comment:
why choose 5? Can it be a static variable?

Reply:
Generally causes are repeated after this depth of 5, so checking till this depth is sufficient to determine the cause.